From 95b32b2ccad75c22063523632322fe3a63f8614f Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Tue, 26 Mar 2013 19:22:18 -0700
Subject: [PATCH 02/96] HBASE-5655 Cap space usage of default log4j rolling policy

Reason: Improvement
Author: Himanshu Vashishtha
Ref: CDH-3602 (sub task: CDH-5069)
(cherry picked from commit 6265420895e7130cb44117b180af545bce4311ea)
---
 bin/hbase-daemon.sh   |    2 +-
 conf/hbase-env.sh     |    8 ++++++++
 conf/log4j.properties |   14 ++++++++++++++
 3 files changed, 23 insertions(+), 1 deletions(-)

diff --git a/bin/hbase-daemon.sh b/bin/hbase-daemon.sh
index 569bbb3..7a0b339 100755
--- a/bin/hbase-daemon.sh
+++ b/bin/hbase-daemon.sh
@@ -116,7 +116,7 @@ fi
 JAVA=$JAVA_HOME/bin/java
 export HBASE_LOG_PREFIX=hbase-$HBASE_IDENT_STRING-$command-$HOSTNAME
 export HBASE_LOGFILE=$HBASE_LOG_PREFIX.log
-export HBASE_ROOT_LOGGER="INFO,DRFA"
+export HBASE_ROOT_LOGGER=${HBASE_ROOT_LOGGER:-"INFO,RFA"}
 export HBASE_SECURITY_LOGGER="INFO,DRFAS"
 logout=$HBASE_LOG_DIR/$HBASE_LOG_PREFIX.out  
 loggc=$HBASE_LOG_DIR/$HBASE_LOG_PREFIX.gc
diff --git a/conf/hbase-env.sh b/conf/hbase-env.sh
index da53a27..497e048 100644
--- a/conf/hbase-env.sh
+++ b/conf/hbase-env.sh
@@ -104,3 +104,11 @@ export HBASE_OPTS="-XX:+UseConcMarkSweepGC"
 
 # Tell HBase whether it should manage it's own instance of Zookeeper or not.
 # export HBASE_MANAGES_ZK=true
+# The default log rolling policy is RFA, where the log file is rolled as per the size defined for the 
+# RFA appender. Please refer to the log4j.properties file to see more details on this appender.
+# In case one needs to do log rolling on a date change, one should set the environment property
+# HBASE_ROOT_LOGGER to "<DESIRED_LOG LEVEL>,DRFA".
+# For example:
+# HBASE_ROOT_LOGGER=INFO,DRFA
+# The reason for changing default to RFA is to avoid the boundary case of filling out disk space as 
+# DRFA doesn't put any cap on the log size. Please refer to HBase-5655 for more context.
diff --git a/conf/log4j.properties b/conf/log4j.properties
index 24d7b2b..2f3bb51 100644
--- a/conf/log4j.properties
+++ b/conf/log4j.properties
@@ -26,6 +26,20 @@ log4j.appender.DRFA.layout=org.apache.log4j.PatternLayout
 # Pattern format: Date LogLevel LoggerName LogMessage
 log4j.appender.DRFA.layout.ConversionPattern=%d{ISO8601} %p %c: %m%n
 
+# Rolling File Appender properties
+hbase.log.maxfilesize=256MB
+hbase.log.maxbackupindex=20
+
+# Rolling File Appender
+log4j.appender.RFA=org.apache.log4j.RollingFileAppender
+log4j.appender.RFA.File=${hbase.log.dir}/${hbase.log.file}
+
+log4j.appender.RFA.MaxFileSize=${hbase.log.maxfilesize}
+log4j.appender.RFA.MaxBackupIndex=${hbase.log.maxbackupindex}
+
+log4j.appender.RFA.layout=org.apache.log4j.PatternLayout
+log4j.appender.RFA.layout.ConversionPattern=%d{ISO8601} %p %c: %m%n
+
 # Debugging Pattern format
 #log4j.appender.DRFA.layout.ConversionPattern=%d{ISO8601} %-5p %c{2} (%F:%M(%L)) - %m%n
 
-- 
1.7.0.4

